pkg/networkservice/chains/nsmgr/heal_test.go

```diff
@@ -346,8 +346,9 @@ func TestNSMGR_CloseHeal(t *testing.T) {
 	conn, err = nsc.Request(ctx, request.Clone())
 	require.NoError(t, err)
 
-	// 3. Stop endpoint
+	// 3. Stop endpoint and wait for the heal to start
 	nseCtxCancel()
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Why do we need to add sleep here?

Reply:
This test validates that heal can be stopped after it has been started - so we need to wait for some time to make sure that heal is really running at the moment when we are trying to stop it.
Without this sleep test is also working, but it tests preventive heal stop - and so it is not the thing that should be tested here.

Reply:
Is there some other way we can make sure that heal has started?
[Recently](https://github.com/networkservicemesh/sdk/pull/989) we found out that on CI 100ms sometimes isn't enough for thread synchronization. And the test where we had issues with timeout is much simpler, so complex heal tests likely require even more time to be sure that another thread started working.

Reply:
This can be problematic.
Waiting for `100ms` doesn't **guarantee** that heal has been started, but it does mean that **in most test runs it will be so**.